### DISCUSSION TITLE
How can I pass a filter value from a filter box to a SQL query Group By clause

### DISCUSSION THREAD

**Discussion by swaza** [apache/superset#15217] [General]
I have a superset dashboard where in i have a filter box containing the name of various dimensions such as area, district, country and so on. I want to generate a run time query to compute the value of a specific metric group by each of these dimensions. The user will select a value of the name of each dimension(column)  from the filter box and the dash should run the query for example:

```
SELECT` 
     SUM(REVENUE),
     {{dim}}
FROM mytable 
GROUP BY {{dim}}
WHERE dim can be 'area' or 'district' or 'country' 
```

How can I achieve this using Jinja templating ? Looking for any examples that would be helpfull ?

**Comment by kamalkeshavani-aiinside** [comment]
This may be useful:
https://github.com/apache/superset/discussions/15272#discussioncomment-896182
